cmd/kubelet/app/server.go

```diff
@@ -1260,6 +1260,10 @@ func RunKubelet(ctx context.Context, kubeServer *options.KubeletServer, kubeDeps
 
 	// process pods and exit.
 	if runOnce {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.LegacyNodeRunOnceMode) {
+			panic("Cannot run runonce mode when LegacyNodeRunOnceMode feature gate is disabled")
```

Comment:
panic will dump a backtrace - is that really what we need?  It will appear to users as a bug or crash.

Reply:
It's not uncommon for `panic` to occur in kubelet, a similar use cases like: https://github.com/kubernetes/kubernetes/blob/0caeba5cbe7793d09586556e056b08f4a8e0adba/pkg/kubelet/status/status_manager.go#L199-L207

Reply:
yeah I agree that panic is a bit much. Can you exist with an error and log an error instead? Personally I think panics should be reserved for coding errors, whereas this is a configuration error.